router/batchrouter/batchrouter.go

```diff
@@ -1831,12 +1830,10 @@ func (worker *workerT) workerProcess() {
 						brt.setJobStatus(batchJob, true, output.Error, postToWarehouseErr)
 						misc.RemoveFilePaths(output.LocalFilePaths...)
 					}
-					destUploadStat.End()
 				case misc.Contains(asyncDestinations, brt.destType):
 					destUploadStat := stats.Default.NewStat(fmt.Sprintf(`batch_router.%s_dest_upload_time`, brt.destType), stats.TimerType)
-					destUploadStat.Start()
+					defer destUploadStat.RecordDuration()()
```

Comment:
it would me more appropriate to not use a defer here and clearly define the timer's start and end
```
start := time.Now()
...
stat.Since(start)
```